File: WorkManager/Data/Entities/Account.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WorkManager.Data.Entities
{
    public class Account
    {
        public int Id { get; set; }
        public string UserName { get; set; }
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Role { get; set; } = 0;
    }
}

File: WorkManager/ClientApp/src/api/account.ts
import { HttpResponseCallback, post } from '../helpers/api';

interface LoginReqestDto {
  userName: string;
  password: string;
}

interface RegisterRequestDto {
  firstName: string;
  lastName: string;
  userName: string;
  password: string;
}

export interface Account {
  userName: string;
  firstName: string;
  lastName: string;
  role: number;
}

export const login = async (
  requestDto: LoginReqestDto,
  callback: HttpResponseCallback<void>
) => post<void>('authorization/login', requestDto, callback);

export const register = async (
  requestDto: RegisterRequestDto,
  callback: HttpResponseCallback<void>
) => post<void>('authorization/register', requestDto, callback);
